lesson-3/station.rb

```diff
@@ -32,6 +32,10 @@ def valid?
     false
   end
 
+  def each_train
+    trains.each { |t| yield(t) if block_given? }
```

Comment:
Not sure that we need `if block_given?` checking. Maybe it would be better to raise an exception.